cubano-concordion/src/main/java/org/concordion/cubano/framework/ConcordionBase.java

```diff
@@ -103,6 +111,30 @@ public Browser getBrowser(String key, BrowserProvider browserProvider) {
         return browsers.get(key);
     }
 
+    private void incrementBrowserTestCount() {
+        if (browserCloseAfterXTests <= 0 || browserTestRunCounted) {
+            return;
+        }
+
+        browserTestRunCounted = true;
+        Integer count = testCount.get();
+
+        if (count >= browserCloseAfterXTests) {
+            Map<String, Browser> browsers = threadBrowsers.get();
```

Comment:
threadBrowsers holds a list of browsers active for the current thread. Generally there would only be one but there have been requests to support multiple browsers for a test.  This will close all browsers on the current thread - not all browsers.